include/kyosu/types/impl/quaternion/axes.hpp

```diff
@@ -0,0 +1,24 @@
+//======================================================================================================================
+/*
+  Kyosu - Complex Without Complexes
+  Copyright : KYOSU Contributors & Maintainers
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+namespace kyosu
+{
+  template < int N > struct axes:std::integral_constant<int, N>{};
+  inline constexpr auto _X(axes<1>{});
+  inline constexpr auto _Y(axes<2>{});
+  inline constexpr auto _Z(axes<3>{});
+
+  template < bool e > struct ext:std::integral_constant<bool, e>{};
+  auto Extrinsic(ext<true>{});
```

Comment:
why not extrinsic and itnrinsic. We have no other markup with capital name

Reply:
these are constexpr values
We can change that later. but we also have _X,_Y and _Z

Reply:
x,y,z are single letter so tha'ts a bit different. I woudl like to discuss the proper naming before merging. LEt's do that tomorrow